src/components/spaces/canvas/index.js

```diff
@@ -86,15 +85,31 @@ export default class CanvasSpace extends Component{
     this.props.dispatch(changeSelect(next))
   }
 
-  renderMetric(metric) {
+  _selectedMetric() {
+   const {selected} = this.props
+   const metrics = _.get(this.props.denormalizedSpace, 'metrics')
+
+   const item = metrics && metrics.filter(i => _.isEqual(i.location, selected))[0];
```

Comment:
Don't need to name item, right? Can just return it directly?

And do we need a defensive check here? Can we push that up higher in the space component, so it short circuits earlier if there are no metrics?